templates/CRM/Contribute/Form/ContributionPage/Settings.tpl

```diff
@@ -10,158 +10,170 @@
-      </tr>
-  {if $contributionPageID}
-        <tr class="crm-contribution-contributionpage-settings-form-block-info_link">
+  </table>
+  <table class="form-layout-compressed">
+    <tr class="crm-contribution-contributionpage-settings-form-block-is_confirm_enabled">
+      <td>&nbsp;</td>
+      <td>{$form.is_confirm_enabled.html} {$form.is_confirm_enabled.label} {help id="id-is_confirm_enabled"}</td>
+    </tr>
+      <tr class="crm-contribution-contributionpage-settings-form-block-is_share">
+      <td>&nbsp;</td>
+      <td>{$form.is_share.html} {$form.is_share.label} {help id="id-is_share"}</td>
+    </tr>
+      <tr class="crm-contribution-contributionpage-settings-form-block-is_active">
+      <td>&nbsp;</td>
+      <td>{$form.is_active.html} {$form.is_active.label}</td>
+    </tr>
+    {if $contributionPageID}
+      <tr class="crm-contribution-contributionpage-settings-form-block-info_link">
         <td>&nbsp;</td>
         <td class="description">
           {if $config->userSystem->is_drupal || $config->userFramework EQ 'WordPress'}
-              {ts}When your page is active, you can link people to the page by copying and pasting the following URL:{/ts}<br />
-              <strong>{crmURL a=1 fe=1 p='civicrm/contribute/transact' q="reset=1&id=`$contributionPageID`"}</strong>
+            {ts}When your page is active, you can link people to the page by copying and pasting the following URL:{/ts}<br />
+            <strong>{crmURL a=1 fe=1 p='civicrm/contribute/transact' q="reset=1&id=`$contributionPageID`"}</strong>
           {elseif $config->userFramework EQ 'Joomla'}
-              {ts 1=$title}When your page is active, create front-end links to the contribution page using the Menu Manager. Select <strong>Administer CiviCRM &raquo; CiviContribute &raquo; Manage Contribution Pages</strong> and select <strong>%1</strong> for the contribution page.{/ts}
+            {ts 1=$title}When your page is active, create front-end links to the contribution page using the Menu Manager. Select <strong>Administer CiviCRM &raquo; CiviContribute &raquo; Manage Contribution Pages</strong> and select <strong>%1</strong> for the contribution page.{/ts}
           {/if}
-  </td>
-  </tr>
-  {/if}
-       </table>
-   <div class="crm-submit-buttons">{include file="CRM/common/formButtons.tpl" location="bottom"}</div>
+        </td>
+      </tr>
+    {/if}
+  </table>
+  {include file="CRM/common/customDataBlock.tpl" customDataType='ContributionPage' customDataSubType=$financialTypeId cid=''}
+  <div class="crm-submit-buttons">{include file="CRM/common/formButtons.tpl" location="bottom"}</div>
 </div>
 
+<script type="text/javascript">
+  CRM.$(function($) {
+    $('#financial_type_id').change( function() {
+      $('#customData_ContributionPage > *').remove();
+      $('#customData_ContributionPage').append(
+        CRM.buildCustomData('ContributionPage', $('#financial_type_id').val(), false, false, false, false, false, false)
```

Comment:
The remove() and append() don't seem needed and I haven't seen those elsewhere. It works with just the buildCustomData line. It uses magic. Dark magic.

Reply:
Even more magic than I thought!  Updated now.